### ISSUE TITLE
Executing wrong Dockerfile

### ISSUE THREAD

**Issue by alvr** [docker/build-push-action#262]
Hello,

I'm having trouble when building an image from a Dockerfile. For example, in [this execution](https://github.com/alvr/alpine-android/runs/1597538541?check_suite_focus=true#step:13:45), the Docker command is wrong, it should be `RUN sdkmanager --sdk_root=/opt/sdk --install "build-tools;27.0.3"` and not `RUN sdkmanager --sdk_root=/opt/sdk --install "build-tools;21.1.2"`.

[Here](https://github.com/alvr/alpine-android/blob/master/android-27/jdk8.Dockerfile) is the Dockerfile I'm using and [here](https://github.com/alvr/alpine-android/blob/master/.github/workflows/default.yaml#L238-L251) the step of the workflow.

Dockerfile:
```dockerfile
FROM alvrme/alpine-android-base:jdk8
LABEL maintainer="Álvaro S. <alvaro@alvr.me>"

ENV BUILD_TOOLS "27.0.3"
ENV TARGET_SDK "27"
ENV PATH $PATH:${ANDROID_SDK_ROOT}/build-tools/${BUILD_TOOLS}

# Install SDK Packages
RUN sdkmanager --sdk_root=${ANDROID_SDK_ROOT} --install "build-tools;${BUILD_TOOLS}" "platforms;android-${TARGET_SDK}" && \
    sdkmanager --sdk_root=${ANDROID_SDK_ROOT} --uninstall emulator
```

Configuration:

```yaml
- name: Build Android API 27 JDK8
  uses: docker/build-push-action@v2
  with:
    context: ./android-27
    file: ./android-27/jdk8.Dockerfile
    no-cache: true
    push: true
    tags: |
      ${{ env.DH_IMAGE_NAME }}:android-27
      ${{ env.DH_IMAGE_NAME }}:android-27-jdk8
      ghcr.io/${{ env.GHCR_IMAGE_NAME }}:android-27
      ghcr.io/${{ env.GHCR_IMAGE_NAME }}:android-27-jdk8
      quay.io/${{ env.QUAY_IMAGE_NAME }}:android-27
      quay.io/${{ env.QUAY_IMAGE_NAME }}:android-27-jdk8
```

In that run, the steps for "Build Android API 21 JDK8" trought "Build Android API 28 JDK8" are having the same problems, all are using the wrong command (except API 21 that is correct)

**Comment by crazy-max** [comment]
@alvr I think you're hitting the same issue as moby/buildkit#1368. In the meantime you could use a [build matrix](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix) like this:

```yml
  build_images_jdk8:
    name: Build Images JDK8
    needs: [build_base_image_jdk8]
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
      matrix:
        android-api:
          - 21
          - 22
          - 23
          - 24
          - 25
          - 26
          - 27
          - 28
          - 29
          - 30
    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v1

      - name: Login to DockerHub
        uses: docker/login-action@v1
        with:
          username: ${{ env.DOCKERHUB_USER }}
          password: ${{ secrets.DOCKERHUB_PASS }}

      - name: Login to GitHub Container Registry
        uses: docker/login-action@v1
        with:
          registry: ghcr.io
          username: ${{ github.repository_owner }}
          password: ${{ secrets.GHCR_PASS }}

      - name: Login to Quay
        uses: docker/login-action@v1
        with:
          registry: quay.io
          username: ${{ env.QUAY_USER }}
          password: ${{ secrets.QUAY_PASS }}

      - name: Build Android API ${{matrix.android-api }} JDK8
        uses: docker/build-push-action@v2
        with:
          context: ./android-${{ matrix.android-api }}
          file: ./android-${{ matrix.android-api }}/jdk8.Dockerfile
          no-cache: true
          push: true
          tags: |
            ${{ env.DH_IMAGE_NAME }}:android-${{ matrix.android-api }}
            ${{ env.DH_IMAGE_NAME }}:android-${{ matrix.android-api }}-jdk8
            ghcr.io/${{ env.GHCR_IMAGE_NAME }}:android-${{ matrix.android-api }}
            ghcr.io/${{ env.GHCR_IMAGE_NAME }}:android-${{ matrix.android-api }}-jdk8
            quay.io/${{ env.QUAY_IMAGE_NAME }}:android-${{ matrix.android-api }}
            quay.io/${{ env.QUAY_IMAGE_NAME }}:android-${{ matrix.android-api }}-jdk8
```

**Comment by alvr** [comment]
Thanks @crazy-max, that worked, but I would also need to add a tag only for one version of the matrix. Is possible to use some kind of condition to add a tag if `matrix.android-api == XX`, like to add `latest` if `android-api` is 30.

Meanwhile I will create a new job only for the latest version.

Thanks again.

**Comment by crazy-max** [comment]
@alvr 

> but I would also need to add a tag only for one version of the matrix.

You can use something like this:

```yml
  build_images_jdk8:
    name: Build Images JDK8
    needs: [build_base_image_jdk8]
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
      matrix:
        android-api:
          - 21
          - 22
          - 23
          - 24
          - 25
          - 26
          - 27
          - 28
          - 29
          - 30
    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Tags
        run: |
          TAGS="android-${{ matrix.android-api }}"
          TAGS="$TAGS,android-${{ matrix.android-api }}-jdk8"
          if [ "${{ matrix.android-api }}" = "30" ]; then
            TAGS="$TAGS,latest"
            TAGS="$TAGS,latest-jdk8"
          fi
          echo "TAGS=$TAGS" >> $GITHUB_ENV

      - name: Docker meta
        id: docker_meta
        uses: crazy-max/ghaction-docker-meta@v1
        with:
          images: |
            ${{ env.DH_IMAGE_NAME }}
            ghcr.io/${{ env.GHCR_IMAGE_NAME }}
            quay.io/${{ env.QUAY_IMAGE_NAME }}
          tag-custom: ${{ env.TAGS }}
          tag-custom-only: true

      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v1

      - name: Login to DockerHub
        uses: docker/login-action@v1
        with:
          username: ${{ env.DOCKERHUB_USER }}
          password: ${{ secrets.DOCKERHUB_PASS }}

      - name: Login to GitHub Container Registry
        uses: docker/login-action@v1
        with:
          registry: ghcr.io
          username: ${{ github.repository_owner }}
          password: ${{ secrets.GHCR_PASS }}

      - name: Login to Quay
        uses: docker/login-action@v1
        with:
          registry: quay.io
          username: ${{ env.QUAY_USER }}
          password: ${{ secrets.QUAY_PASS }}

      - name: Build Android API ${{matrix.android-api }} JDK8
        uses: docker/build-push-action@v2
        with:
          context: ./android-${{ matrix.android-api }}
          file: ./android-${{ matrix.android-api }}/jdk8.Dockerfile
          no-cache: true
          push: true
          tags: ${{ steps.docker_meta.outputs.tags }}
```

**Comment by alvr** [comment]
Thank you very much @crazy-max, it is working flawless.

**Comment by crazy-max** [comment]
Fixed since BuildKit 0.9.0
